cmd/nitro/nitro.go

```diff
@@ -741,10 +757,18 @@ func (c *LiveNodeConfig) Start(ctxIn context.Context) {
 	})
 }
 
+func (c *LiveNodeConfig) setOnReloadHook(hook OnReloadHook) {
+	// setting the hook doesn't block others from reading the config
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	c.onReloadHook = hook
```

Comment:
We should document that this can't be called concurrently and doesn't support having multiple hooks.

Reply:
I'll add the comment. I think that it makes also sense to skip locking the mutex here - seems that there is no need for that as we don't allow  (and don't need) concurrent calls of the method